enjam comments, add op_codes

2qx opened the pull request:
This PR refactors the staking contract to the form:

    // comment
    // OP_code
    // cashScript()

Moving the comments above and adding the OP_code correlate:

- Makes comment's legible to auditors on narrow screens (<100 char).
- Makes the logic legible to auditors unfamiliar with CashScript. 

Minimal files to compile the CashScript with cashc v 0.10 were added.

Commit:
enjam comments, op_code

Commit:
comment unlock token amount constraint

Commit:
change withdraw fn to setLockingFee()

SayoshiNakamario commented:
Awesome work! 
I think the contracts and UTXO mindset is easier to understand in the current layout when considering it from a new-to-coding perspective, but agree your layout is better for experienced users/auditing especially with the included OP codes. 

Since Badgers is partially a 'learning showcase' example I think we should have both layouts, so if you add them as new files I'm happy. 😊

For withdraw() it can be either/or since it's dual-purpose of withdrawing the accumulated sats fee to admins address and optionally setting the locking fee. I'd be fine with it staying as withdraw() on the original version and being setLockingFee() on the refactor. For those that dig into it shows that the function names can be renamed without affecting the compiled contract.

Commit:
add readme, move audit format

Commit:
fix typo

Commit:
revert json

2qx commented:
Okay, I added the other version as BadgerStake.op_codes.cash so if someone prefers that it's there. 

Also added a brief README.md in the style of your CashStarter readme, but shorter.

SayoshiNakamario commented:
Awesome, thanks for the effort!
